### DISCUSSION TITLE
Increasing XC40 build memory

### DISCUSSION THREAD

**Discussion by alanjhewitt** [MetOffice/simulation-systems#224] [LFRic]
I have solved all of the code problems with my 3d UKCA to LFRic branch.

There is one problem remaining, that should be quite trivial for someone else, but I dont know the answer.

I have five xc40 builds that run out of memory:

http://fcm1/cylc-review/taskjobs/ahewitt?&suite=vn1.1_l_a_side_3d%2Frun1&no_fuzzy_time=0&tasks=build_lfric_atm_xc40_intel_production-64bit-rbl32-rtran32

http://fcm1/cylc-review/taskjobs/ahewitt?&suite=vn1.1_l_a_side_3d%2Frun1&no_fuzzy_time=0&tasks=build_lfric_atm_xc40_intel_fast-debug-64bit

http://fcm1/cylc-review/taskjobs/ahewitt?&suite=vn1.1_l_a_side_3d%2Frun1&no_fuzzy_time=0&tasks=build_lfric_atm_xc40_intel_fast-debug-64bit-rbl32-rtran32

http://fcm1/cylc-review/taskjobs/ahewitt?&suite=vn1.1_l_a_side_3d%2Frun1&no_fuzzy_time=0&tasks=build_ngarch_xc40_intel_fast-debug-64bit

http://fcm1/cylc-review/taskjobs/ahewitt?&suite=vn1.1_l_a_side_3d%2Frun1&no_fuzzy_time=0&tasks=build_lfric_coupled_xc40_intel_fast-debug-64bit

I am trying to work out what I need to change to request more memory. It looks like rose-stem/site/meto/common/suite_config_xc40.cylc under [[XC_BUILD]] might be the correct place , because the current setting of 18GB is about the same size as the error message:

============================= PBS epilogue =============================

error: process 'fortcom' killed after using 11825584kB of memory
error: from a total of 18874368kB (63% of usage)
error: overall job memory limit was 18874368kB (100% of limit)
error: job terminated by cgroup memory limit

End of Job Report
Run at 2024-07-23 16:40:24 for job 1459332.xce00

**Comment by james-bruten-mo** [comment]
Yep that's the right place. It can go up to the maximum value for the xc40 shared queue which I'm reliably informed is 118GB.

**Comment by stevemullerworth** [comment]
You really need to investigate what is causing the compiler to run out of memory, otherwise it can become a long term issue to manage - particularly if it is also taking a lot of time. Ideally, code can be modified or compiler options for individual files modified to reduce the compile cost to a manageable amount.

**Reply by alanjhewitt** [reply]
Ian mentioned an Intel compiler bug that has a work around on the trunk, but uses a lot of memory. from Ian:

**essentially there is a bug in the intel compiler that we have a workaround for in place, but the workaround uses a huge amount of memory. the bug only affects openmp in the psy layer, but the whole point of doing this work is so we can use the openmp in the native code and not in the psy-layer. so i've removed the openmp from the psy-layer (what the new file does), and then added an over-ride to turn off the compiler workaround for that file**

So now I have a work around to the workaround to allow the code to build with my changes

https://code.metoffice.gov.uk/trac/lfric_apps/changeset/2746/main/branches/dev/alanjhewitt/vn1.1_l_a_side_3d
